src/extension.ts

```diff
@@ -365,6 +365,10 @@ class ExtensionManager implements vscode.Disposable {
     if (should_configure) {
       // We've opened a new workspace folder, and the user wants us to
       // configure it now.
+      if (cmt.workspaceContext.config.scanForKitsOnOpen) {
+        await this.scanForKits();
```

Comment:
Thinking about this... If we have to scan every time the folder opens, this seems excessive and will slow down IntelliSense for cpptools because the configurations won't be sent until after 'configure' happens.

We should only need to scan when we know something has changed with the Kits, right?  I think we could manage this with a piece of persistent state that tracks a "version number" or something. If we ever make a change that requires a rescan, then we bump the number. If the state doesn't exist, we do a scan (e.g. this will happen for everyone once they upgrade to 1.4, but then it won't happen again unless we change Kits again).  If we take this approach, we don't need a setting to disable it.

Reply:
True, good idea. 
I just pushed an update. Basically I reverted the whole thing. 
I found a better spot to hook the breaking changes scan at the beginning of general activation (it didn't have to be part of the configure process, I was following too closely the setting cmake.configureOnOpen).
I implemented some versioning functionality tied to the extension version defined in package.json.
To avoid yet another version to take care of and also to be able to rescan for kits appropriately when users go back and forth with various versions of CMake Tools I decided to do this rescan every time when the kits version doesn't match the extension version. It will be slightly more often than if we implement a new versioning for kits but I think it's acceptable and avoids too complicated implementation,.